modules/tinymce/src/core/main/ts/api/EditorManager.ts

```diff
@@ -373,19 +374,18 @@ const EditorManager: EditorManager = {
       ' '
     );
 
-    const isInvalidInlineTarget = function (settings, elm) {
-      return settings.inline && elm.tagName.toLowerCase() in invalidInlineTargets;
-    };
+    const isInvalidInlineTarget = (settings: RawEditorSettings, elm: HTMLElement) =>
+      settings.inline && elm.tagName.toLowerCase() in invalidInlineTargets;
 
-    const createId = function (elm) {
+    const createId = function (elm: HTMLElement) {
       let id = elm.id;
 
       // Use element id, or unique name or generate a unique id
       if (!id) {
-        id = elm.name;
+        id = (elm as unknown as HTMLTextAreaElement).name as string | undefined;
 
         if (id && !DOM.get(id)) {
-          id = elm.name;
+          // Keep current name
```

Comment:
This could also be improved to remove the empty block. eg:
```
if (!id || DOM.get(id)) {
  // Generate unique name
  id = DOM.uniqueId();
}
```

Reply:
Another option, which would potentially be even better, would be to do something like this:
```
id = Obj.get(elm, 'name').filter((name) => !DOM.get(name)).getOrThunk(DOM.uniqueId);
```